### PR TITLE
GAT-1490 change ping logic

### PR DESCRIPTION
**Pull request by johnm7115**

### CODE REVIEW

File: org.eclipse.paho.client.mqttv3/src/main/java/org/eclipse/paho/client/mqttv3/internal/ClientState.java [hunk 1]
```diff
@@ -698,49 +676,34 @@ public MqttToken checkForActivity(IMqttActionListener pingCallback) throws MqttE
...
-
-                    //Wake sender thread since it may be in wait state (in ClientState.get())                                                                                                                             
-                    notifyQueueLock();
-                }
-                else {
-                    log.fine(CLASS_NAME, methodName, "634", null);
-                    nextPingTime = Math.max(1, getKeepAlive() - (time - lastOutboundActivity));
-                }
+                // Always send a ping if we get here
+				//@TRACE 620=ping needed. keepAlive={0} lastOutboundActivity={1} lastInboundActivity={2}
```
Comment (dmather) [inline org.eclipse.paho.client.mqttv3/src/main/java/org/eclipse/paho/client/mqttv3/internal/ClientState.java]: What's with the indent here?

File: org.eclipse.paho.client.mqttv3/src/main/java/org/eclipse/paho/client/mqttv3/internal/ClientState.java [hunk 2]
```diff
@@ -698,49 +676,34 @@ public MqttToken checkForActivity(IMqttActionListener pingCallback) throws MqttE
...
+					token.setActionCallback(pingCallback);
+				}
+				tokenStore.saveToken(token, pingCommand);
+				pendingFlows.insertElementAt(pingCommand, 0);
+
+				//Wake sender thread since it may be in wait state (in ClientState.get())
+				notifyQueueLock();
+
+				// Schedule the next ping at half the keep alive interval to ensure we don't get disconnected
+				nextPingTime = getKeepAlive() / 2;
```
Comment (dmather) [inline org.eclipse.paho.client.mqttv3/src/main/java/org/eclipse/paho/client/mqttv3/internal/ClientState.java]: This all seems reasonable.

**Review by dmather — COMMENTED** [review]
We'll want to figure out versioning.

File: org.eclipse.paho.client.mqttv3/src/main/java/org/eclipse/paho/client/mqttv3/internal/ClientState.java (hunk 2 above)
Comment (johnm7115) [reply org.eclipse.paho.client.mqttv3/src/main/java/org/eclipse/paho/client/mqttv3/internal/ClientState.java]: hmm, looks like the file is a mix of tabs and spaces.

**Review by dmather — APPROVED** [review]
Everything seems reasonable here.

**Pull request merged**